include/seqan3/io/detail/parse_condition.hpp

```diff
@@ -0,0 +1,390 @@
+    }
+};
+
+//!\todo write me
+template <char rng_beg, char rng_end>
+using is_in_range = parse_condition<is_in_range_impl<rng_beg, rng_end>>;
+
+//!\todo write me
+template <alphabet_concept alphabet_t>
+struct is_in_alphabet_impl
+{
+    //!\todo write me
+    static constexpr auto msg = constexpr_string{"is_in_alphabet<"} +
+                                               constexpr_string{detail::get_display_name_v<alphabet_t>} +
+                                               constexpr_string{">"};
+    //!\todo write me
+    template <typename value_t>
+    bool operator()(value_t && val) const
+    //\cond
+        requires alphabet_concept<std::decay_t<value_t>>
+    //\endcond
+    {
+        using seqan3::to_char;
+        using seqan3::assign_char;
+
+        if constexpr (std::is_same_v<std::decay_t<value_t>, std::decay_t<alphabet_t>>)
+            return true;
+
+        alphabet_t cmp;
+        assign_char(cmp, to_char(val));
+        return to_char(cmp) == std::toupper(static_cast<uint8_t>(to_char(val)));
+    }
+};
+
+//!\todo write me
+template <alphabet_concept alphabet_t>
+using is_in_alphabet = parse_condition<is_in_alphabet_impl<alphabet_t>>;
+
+//!\todo write me
+template <char char_v>
+struct is_char_impl
+{
+    //!\todo write me
+    static constexpr auto msg = constexpr_string{"is_char<"} +
+                                               constexpr_string{char_v} +
+                                               constexpr_string(">");
+
+    //!\todo write me
+    template <typename value_t>
+    bool operator()(value_t && val) const
+    //\cond
+        requires alphabet_concept<std::decay_t<value_t>>
+    //\endcond
+    {
+        return to_char(val) == char_v;
+    }
+};
+
+//!\todo write me
+template <char char_v>
+using is_char = parse_condition<is_char_impl<char_v>>;
+
+// ----------------------------------------------------------------------------
+// Composite Functors
+// ----------------------------------------------------------------------------
+// Don't use isblank() or isspace() as it they seem to be slower than our functors (due to inlining)
+
+//!\todo write me
+constexpr is_char<'\t'> is_tab;
+//!\todo write me
+constexpr is_char<' '> is_space;
+//!\todo write me
+constexpr decltype(is_space || is_tab) is_blank;
+//!\todo write me
+constexpr decltype(is_char<'\n'>{} || is_char<'\r'>{}) is_new_line;
+//!\todo write me
+constexpr parse_condition<or_fn<is_char<' '>, is_char<'\t'>, is_char<'\n'>, is_char<'\r'>>> is_whitespace ;
+//!\todo write me
+constexpr is_in_range<'!', '~'> is_graph;
+//!\todo write me
+constexpr decltype(is_in_range<'a', 'z'>{} || is_in_range<'A', 'Z'>{}) is_alpha;
+//!\todo write me
+constexpr is_in_range<'0', '9'> is_digit;
+//!\todo write me
+constexpr parse_condition<or_fn<is_in_range<'a','z'>, is_in_range<'A','Z'>, is_in_range<'0','9'>>> is_alpha_num;
+
+//!\todo write me
+template <parse_condition_concept condition_type, typename error_type = std::runtime_error>
```

Comment:
Why do you use `error_type` as template and not a fixed `runtime_error`? Template does not seem necessary.